### ISSUE TITLE
In `React Example: With Trpc React Query` the output type infer doesn't work

### ISSUE THREAD

**Issue by michaltaberski** [TanStack/router#3057]
### Which project does this relate to?

Router

### Describe the bug

Using `trpc` together with `react-query` in the context of react router is making the procedure output type to infer to `any`. The issue is visible in TanStack/Router example code.

### Your Example Website or App

https://tanstack.com/router/latest/docs/framework/react/examples/with-trpc-react-query

### Steps to Reproduce the Bug or Issue

1. Go to [React Example: With Trpc React Query](https://tanstack.com/router/latest/docs/framework/react/examples/with-trpc-react-query)
2. Open `dashboard.index.tsx` file
3. Check notation of the `posts` object. It doesn't infer output type from the trcp prodecure:
![Image](https://github.com/user-attachments/assets/cbe66e6b-5a74-4c0b-bcd5-bbcc828aa171)

#### Aside notes:
I am experiencing the same issue and at first I thought it's my fault, but after a bit of exploring I know it's a common issue. For reference, I have asked for a help with that issue a while ago on trpc repo: https://github.com/trpc/trpc/discussions/6314.

### Expected behavior

The procedure output type should be inffered from `trpc`.

### Platform

- Browser: Arc
- Version: Any

**Comment by schiller-manuel** [comment]
seems to be a TRPC issue and not router related, right?

**Comment by michaltaberski** [comment]
Yes, it is. We can close this ticket, but for those who are interested in fix (or "semi-fix") - the solution that works for me is to enhance trpc init with transformer, like:

```js
export const t = initTRPC.create({
  transformer: superjson,
});
```

in theory it should not be needed to have types properly working, but in practice it solved my issue 🤔 .
